src/mono/wasm/debugger/BrowserDebugProxy/DevToolsHelper.cs

```diff
@@ -61,6 +61,8 @@ internal class DotnetObjectId
     {
         public string Scheme { get; }
         public string Value { get; }
+        public string SubScheme { get; }
```

Comment:
I think it would be useful to add a deconstructor which returns `int` values?
https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct#user-defined-types

```sharp
public void Deconstruct(out string scheme, out int value, out string subscheme, out int? subvalue)
{..}
```

.. and use it as (example):
```csharp
var (_, value, subscheme, _) = dotnetObjectId;
```

`int value` - so the callers don't have to worry about `int.Parse` failing. And IIUC, the `string value` should always parse as an int.

And `int? subValue as a `null` value would indicate a missing value.

Reply:
Also, we should maybe use an enum for this, even if it has only one value `MethodId` to start with. That would avoid callers having to check for `== "methodId"`, and having any typos.